pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -268,6 +273,53 @@ func (c *InstallerController) manageInstallationPods(operatorSpec *operatorv1.Op
 	return false, nil
 }
 
+func (c *InstallerController) updateRevisionStatus(operatorStatus *operatorv1.StaticPodOperatorStatus) error {
+	failedRevisions := make(map[int32]struct{})
+	currentRevisions := make(map[int32]struct{})
+	for _, nodeState := range operatorStatus.NodeStatuses {
+		failedRevisions[nodeState.LastFailedRevision] = struct{}{}
+		currentRevisions[nodeState.CurrentRevision] = struct{}{}
+	}
+
+	// If all current revisions point to the same revision, then mark it successful
+	if len(currentRevisions) == 1 {
+		for currentRevision := range currentRevisions {
+			err := c.updateConfigMapForRevision(currentRevision, string(corev1.PodSucceeded))
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	for failedRevision := range failedRevisions {
+		if failedRevision == 0 {
+			continue
+		}
+		err := c.updateConfigMapForRevision(failedRevision, string(corev1.PodFailed))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *InstallerController) updateConfigMapForRevision(currentRevision int32, phase string) error {
+	statusConfigMap, err := c.kubeClient.CoreV1().ConfigMaps(c.targetNamespace).Get(statusConfigMapNameForRevision(currentRevision), metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	statusConfigMap.Data["revision"] = fmt.Sprintf("%d", currentRevision)
```

Comment:
I like having something in the map with the revision ID better than trying to parse it out of the name, as done here: https://github.com/openshift/library-go/blob/master/pkg/operator/staticpod/controller/prune/prune_controller.go#L95

Actually this also goes back to the case for using a label, or some sort of grouping, because we're already parsing the name just to see if we have the right configmap: https://github.com/openshift/library-go/blob/master/pkg/operator/staticpod/controller/prune/prune_controller.go#L91

Which makes it harder to check for `my-special-map` vs `my-special-map-other`, and would also make selecting easier. Wdyt?

Reply:
It might be more reliable to set this specific entry here though https://github.com/openshift/library-go/pull/157/files#diff-288bef64e55c090483c4a0afb1145359R167